VeganVirus/VeganVirus/SoundAction.cpp

```diff
@@ -2,42 +2,46 @@
 
 SoundAction::SoundAction(double req) : Action(req, MUSIC_ICON_PATH)
 {
-    playListTime[0] = 62;
+    _timer = 0;
 }
 
 void SoundAction::start()
 {
-    if (!canActivate())
+    _timer = SOUND_DUARTION;
+}
+
+void SoundAction::update(double dt)
+{
+    std::string song = "";
+    _timer -= dt;
+    if (_timer <= 0 || !canActivate())
     {
         return;
     }
-    std::string song = "";
+    
     //change the volume to 30 for now
-    this->_audioManager.changeVolume(30);
+    this->_audioManager.changeVolume(60);
 
     //choose randomly song
     this->currentSong = rand() % NUM_OF_SONGS;
     song = "play music\\song" + std::to_string(this->currentSong) + ".wav";
 
     //play and check when song start
     this->_audioManager.play(song);
-    this->songStartTime = clock();
 }
 
 
 bool SoundAction::canActivate()
 {
-    //check that song is playing
+    double timePass = SOUND_DUARTION - _timer;
+    
+    //check that song no is playing
     if (this->currentSong != -1)
     {
-        //substract the current time by the start time
-        clock_t duration = clock() - this->songStartTime;
-        //convert ms to s
-        duration /= 1000;
-        if (duration < playListTime[currentSong])
+        if (timePass <= playTimeList[currentSong])
         {
             return false;
-        }
+        }  
     }
     return true;
```

Comment:
This can be simplified to a simple condition.
First, I suggest incrementing `_timer` starting from 0 instead of decrementing it. Doing `_timer += dt;`

Than, simplify this method to:
`return this->_currentSong == NO_SONG || _timer > playTimeList[_currentSong]`.

Also, you should add a `#define` to the value of -1 you use.
Maybe `#define NO_SONG -1`.